indexer.go

```diff
@@ -0,0 +1,31 @@
+package indexer
+
+// Provider interface of needed provider functions
+type Provider interface {
+	GetBlockHeight() (int, error)
+	// TODO: add missing functions
+}
+
+// Database interface of needed database functions
+type Database interface {
+	// TODO: add missing functions
+}
+
+// Indexer struc handler for Indexer functions
+type Indexer struct {
+	provider Provider
+	database Database
+}
+
+// NewIndexer returns Indexer instance with given input
+func NewIndexer(provider Provider, database Database) *Indexer {
+	return &Indexer{
+		provider: provider,
+		database: database,
+	}
+}
+
+// IndexBlock transfers all transactions to a known strcuture and saves them
+func (i *Indexer) IndexBlock(block int) error {
```

Comment:
`IndexBlock` name and a `block` argument makes the naming redundant, could be simply called index and you'll know is a block by the argument

Reply:
The param should be called `block_number`

Reply:
`block_number` is also redundant and using pascal case, we know block is of type number so no need to add it to the name

Reply:
Pascal case slipped thru there, so dismiss that. 

Thing is that I don't believe it is redundant. A block is more complex than a number, even if the types specifies that it should be a number. We should use the appropriate term for the argument, which is a given block _number_.

This is `pocket-go` code, and I think that's totally fine.
```
func (p *Provider) GetBlock(blockNumber int) (*GetBlockOutput, error) {}
```

Reply:
If y'all disagree about the naming, the _real_ and correct name is `blockHeight`, as that is the name used on `pocket-core`. I do think we should keep the `block` in the name.

Reply:
> If y'all disagree about the naming, the _real_ and correct name is `blockHeight`, as that is the name used on `pocket-core`. I do think we should keep the `block` in the name. Not everyone seeing this code will have so much context on what this is doing.

Agreed